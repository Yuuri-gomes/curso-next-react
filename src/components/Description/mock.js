export const DescriptionPropsMock = {
  description: "dummy description",
};
