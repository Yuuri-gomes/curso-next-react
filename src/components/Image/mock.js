export const ImagePropsMock = {
  cover: "img/img.jpg",
  title: "title 1",
};
