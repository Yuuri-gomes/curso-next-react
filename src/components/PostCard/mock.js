export const PostCardPropsMock = {
  title: "title 1",
  description: "Body 1",
  id: 1,
  cover: "img/img.png",
};
