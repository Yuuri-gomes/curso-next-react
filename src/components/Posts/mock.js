export const PostsPropsMock = {
  posts: [
    {
      id: 1,
      title: "Title 1",
      body: "body 1",
      cover: "img/img.jpg",
    },
    {
      id: 2,
      title: "Title 2",
      body: "body 2",
      cover: "img/img.jpg",
    },
    {
      id: 3,
      title: "Title 3",
      body: "body 3",
      cover: "img/img.jpg",
    },
  ],
};
