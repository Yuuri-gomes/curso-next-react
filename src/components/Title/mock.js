export const TitlePropsMock = {
  title: "Some title",
};
