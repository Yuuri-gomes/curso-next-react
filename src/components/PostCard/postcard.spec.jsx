import { render, screen } from "@testing-library/react";
import { PostCardComponent } from ".";
import { PostCardPropsMock } from "./mock";

describe("<PostCard />", () => {
  it("should render PostCard correctly", () => {
    render(<PostCardComponent {...PostCardPropsMock} />);

    expect(screen.getByAltText(PostCardPropsMock.title)).toHaveAttribute(
      "src",
      PostCardPropsMock.cover
    );

    expect(
      screen.getByRole("heading", { name: PostCardComponent.title })
    ).toBeInTheDocument();

    expect(screen.getByText(PostCardPropsMock.description)).toBeInTheDocument();
  });

  it("Should match snapshot", () => {
    const { container } = render(<PostCardComponent {...PostCardPropsMock} />);

    expect(container.firstChild).toMatchSnapshot();
  });
});
