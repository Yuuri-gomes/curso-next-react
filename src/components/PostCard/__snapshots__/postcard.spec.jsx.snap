// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PostCard /> Should match snapshot 1`] = `
<div
  class="post-card"
>
  <img
    alt="title 1"
    src="img/img.png"
  />
  <div
    class="post-content"
  >
    <h1>
      title 1
       
      1
    </h1>
    <p>
      Body 1
    </p>
  </div>
</div>
`;
